### PR TITLE
On Red Hat systems, add necessary kernel command line options

### PR DESCRIPTION
**Pull request by ajdecon**
## Summary

For Enroot to work on RHEL-derived systems, the [kernel command line needs to be configured](https://github.com/NVIDIA/enroot/blob/master/doc/requirements.md#kernel-command-line) to correctly enable user namespaces.

This PR enables these kernel command line options if necessary when running on RHEL systems, and reboots the node if we made any changes.

We also add a boolean flag `enroot_configure_kernel_options` that allows us to disable this behavior.

## Test plan

Set up a new CentOS VM and check its kernel command line:

```
[vagrant@localhost ~]$ cat /proc/cmdline
BOOT_IMAGE=/boot/vmlinuz-3.10.0-957.12.2.el7.x86_64 root=UUID=8ac075e3-1124-4bb6-bef7-a6811bf8b870 ro no_timer_check console=tty0 console=ttyS0,115200n8 net.ifnames=0 biosdevname=0 elevator=noop crashkernel=auto LANG=en_US.UTF-8
```

Run the playbook:

```
$ cat playbooks/container/enroot.yml
---
- hosts: all
  become: yes
  roles:
  - ansible-role-enroot

$ ansible-playbook -i virtual/config/inventory -l virtual-mgmt01 playbooks/container/enroot.yml
```

After it completes, note that the node was rebooted:

```
RUNNING HANDLER [ansible-role-enroot : reboot node] *************************************************
changed: [virtual-mgmt01]
```

And check the command line:

```
[vagrant@localhost ~]$ cat /proc/cmdline
BOOT_IMAGE=/boot/vmlinuz-3.10.0-957.12.2.el7.x86_64 root=UUID=8ac075e3-1124-4bb6-bef7-a6811bf8b870 ro no_timer_check console=tty0 console=ttyS0,115200n8 net.ifnames=0 biosdevname=0 elevator=noop crashkernel=auto LANG=en_US.UTF-8 namespace.unpriv_enable=1 user_namespace.enable=1
```

Re-running the playbook correctly skips making changes to the kernel command line.

```
TASK [ansible-role-enroot : check if kernel unpriv enabled] *****************************************
changed: [virtual-mgmt01]

TASK [ansible-role-enroot : check if user namespaces enabled] ***************************************
changed: [virtual-mgmt01]

TASK [ansible-role-enroot : install grubby if needed] **********************************************
skipping: [virtual-mgmt01]

TASK [ansible-role-enroot : add kernel options to use enroot] ****************************************
skipping: [virtual-mgmt01] => (item=namespace.unpriv_enable=1)
skipping: [virtual-mgmt01] => (item=user_namespace.enable=1)
```

### CODE REVIEW

**Comment by supertetelman** [comment]
Can you revert this commit to add the enroot tests back? 

https://github.com/NVIDIA/deepops/commit/d13c3c7b1295a1aad53e0af7ff324d3380dfde07

**Comment by ajdecon** [comment]
Please don't merge this right away -- the command line seems to be correct, but my enroot tests are failing. Will update when fixed

**Comment by ajdecon** [comment]
OK, this is ready to review. The sysctl settings seem to have been the necessary bits. The following now works on my VM test cluster:

```
[vagrant@virtual-login01 ~]$ srun -n1 cat /etc/centos-release
CentOS Linux release 7.6.1810 (Core)
[vagrant@virtual-login01 ~]$ srun -n1 --container-image="centos:8" cat /etc/centos-release
pyxis: importing docker image ...
pyxis: creating container filesystem ...
pyxis: starting container ...
CentOS Linux release 8.3.2011
```

File: tasks/redhat.yml [hunk 1]
```diff
@@ -35,3 +35,40 @@
...
+  failed_when: kernel_user_namespace.rc == 127
+
+- name: install grubby if needed
+  yum:
+    name: "grubby"
+    state: "present"
+  when: ((kernel_unpriv_enable.rc == 1) or (kernel_user_namespace.rc == 1)) and enroot_configure_kernel_options
+
+- name: add kernel options to use enroot
+  command: "grubby --update-kernel={{ enroot_configure_kernel_options_versions }} --args={{ item }}"
```
Comment (lukeyeager) [inline tasks/redhat.yml:56]: On Ubuntu I find it's cleanest to add drop-in files to `/etc/default/grub.d/`. Is the recommended method on EL systems?

File: tasks/redhat.yml [hunk 2]
```diff
@@ -35,3 +35,40 @@
...
+  when: ((kernel_unpriv_enable.rc == 1) or (kernel_user_namespace.rc == 1)) and enroot_configure_kernel_options
```
Comment (lukeyeager) [inline tasks/redhat.yml:57]: Consider using a block for these two tasks to avoid copying this complex if condition? Not a blocker - just a protip.

**Review by lukeyeager — APPROVED** [review]
Added a few nitpicks - feel free to ignore them. Thanks, this is useful!

File: tasks/redhat.yml (hunk 1 above)
Comment (ajdecon) [reply tasks/redhat.yml:56]: Yes, grubby seems to be recommended in most of the RHEL docs I could find.

**Pull request merged**